### PR TITLE
Performance improvements: avoid boxing and needless string to name conversion

### PR DESCRIPTION
**Pull request by retronym**

### CODE REVIEW

**Comment by retronym** [comment]
Simplifications to dead state elimination and generation of .dot visualiations of the state machine moved to #198

**Comment by retronym** [comment]
Superceded by #194

File: src/main/scala/scala/async/internal/AsyncMacro.scala [hunk 1]
```diff
@@ -1,10 +1,19 @@
 package scala.async.internal
 
+import java.util
+
 object AsyncMacro {
+  private val nameCache = new util.WeakHashMap[Object, AsyncNames[_]]()
   def apply(c0: reflect.macros.Context, base: AsyncBase)(body0: c0.Tree): AsyncMacro { val c: c0.type } = {
     import language.reflectiveCalls
+    val asyncNames0 = nameCache.synchronized[AsyncNames[_]] {
```
[the remark this answers is not in the document]
Comment (retronym) [reply src/main/scala/scala/async/internal/AsyncMacro.scala]: I got rid of this map altogether #194 as I think this was leaky anyway.

**Comment by viktorklang** [comment]
@retronym Err, #198 you mean?

**Pull request closed without merging**